### ISSUE TITLE
Paginator: jumpToPageDropdown not working with Pages greater than 10

### ISSUE THREAD

**Issue by hb237** [primefaces/primefaces#5762]
# 1) Environment
- Does not work on PrimeFaces version: 8.0
- Did work before on PrimeFaces version: 6.2
- Application server + version: Tomcat v8.0
- Affected browsers: Chrome 80, Firefox 74, Internet Explorer 11

## 2) Expected behavior
The paginator should be able to display pages that have a number greater than 10.

## 3) Actual behavior
When clicking on a a number greater than ten in the paginator or the jumpToPageDropdown the results still get displayed one last time, but then the page is locked and no interaction can be performed anymore. 

The following error is shown:
```
jquery.js.xhtml?ln=primefaces&v=8.0:2 Uncaught Error: Syntax error, unrecognized expression: option[value=\31 0]
    at Function.se.error (jquery.js.xhtml?ln=primefaces&v=8.0:2)
    at se.tokenize (jquery.js.xhtml?ln=primefaces&v=8.0:2)
    at se.compile (jquery.js.xhtml?ln=primefaces&v=8.0:2)
    at se.select (jquery.js.xhtml?ln=primefaces&v=8.0:2)
    at se (jquery.js.xhtml?ln=primefaces&v=8.0:2)
    at Function.se.matches (jquery.js.xhtml?ln=primefaces&v=8.0:2)
    at Function.k.filter (jquery.js.xhtml?ln=primefaces&v=8.0:2)
    at k.fn.init.k.fn.<computed> [as children] (jquery.js.xhtml?ln=primefaces&v=8.0:2)
    at c.updateUI (components.js.xhtml?ln=primefaces&v=8.0:58)
    at c.updateTotalRecords (components.js.xhtml?ln=primefaces&v=8.0:58)
```

## 4) Steps to reproduce

Open the page with the data table and click on any page number greater than 10 or select it from the . Remove {JumpToPageDropdown} from the paginatorTemplate and the error will not show up anymore. Also bigger page numbers than 10 can now be selected. 

## 5) Sample XHTML and Sample bean

This showcase: https://www.primefaces.org/showcase/ui/data/datatable/paginator.xhtml

Modify the datatable in paginator.xml with this snippet:
```
<p:dataTable var="car" value="#{dtPaginatorView.cars}" rows="1"
   paginator="true" paginatorPosition="top" pageLinks="15"
   paginatorTemplate="{CurrentPageReport} {FirstPageLink} {PreviousPageLink} {PageLinks} {JumpToPageDropdown} {NextPageLink} {LastPageLink}"
   currentPageReportTemplate="{startRecord}-{endRecord} of {totalRecords} records">
```

## 6) Discovered bug:

I already explained the bug here in depth: 

https://stackoverflow.com/questions/60989240/primefaces-paginator-jumptopagedropdown-bug

#### Location of error:
File: `components.js.xhtml?ln=primefaces&v=8.0`
Widget: `PrimeFaces.widget.Paginator`
Function: `updateUI`
Line: `this.jtpSelect.children("option[value=" + $.escapeSelector(this.cfg.page) + "]").prop("selected", "selected")`

#### Explanation:
`$.escapeSelector()` is not needed for numbers and produces faulty outputs. The call of `this.jtpSelect.children()` in turn causes an error due to a wrong parameter.

`this.cfg.page` uses zero-based numbering. Page 1 is 0, page 10 is 9 and so on. In version 6.2 `this.jtpSelect.children()` would be called with option[value=10] when the selected page is 11. But in version 8.0 `$.escapeSelector()` would format the same input to `\31 0` and results in `option[value=\31 0]`.

#### Error message:
The part _option[value=\31 0]_ shows, that the function escapeSelector produces useless outputs. It is simply not needed here.

```
jquery.js.xhtml?ln=primefaces&v=8.0:2 Uncaught Error: Syntax error, unrecognized expression: option[value=\31 0]

```

### Suggested fix:
In my Stack Overflow post I have provided an answer how the method should look like, to fix the issue. I already applied this hack in my project by overriding `window.PrimeFaces.widget.Paginator.prototype.updateUI` with the proposed function. Afterwards I could use the jumpToPageDropdown as before without any issues. I would have provided a PR for this issue, but I did not find the file where the change has to be made.

**Comment by melloware** [comment]
The location of the file you need to edit for the PR is `\src\main\resources\META-INF\resources\primefaces\paginator\paginator.js`

**Comment by hb237** [comment]
Here is the PR: https://github.com/primefaces/primefaces/pull/5763

**Comment by hb237** [comment]
Will this issue still be fixed in v8.0 or later in v9.0?

**Comment by melloware** [comment]
Hopefully @mertsincan will merge it into Elite 8.0.1 release.
